File: src/main/java/de/byjoker/jfql/connection/Connection.java
package de.byjoker.jfql.connection;

import de.byjoker.jfql.util.Result;
import de.byjoker.jfql.util.User;

public interface Connection {

    void connect();

    void connect(String host, User user);

    boolean isConnected();

    void disconnect();

    Result query(String query);

    Result query(String query, boolean exceptions);

    Result query(String query, Object... replacers);

    Result query(String query, boolean exception, Object... replacers);

    String stringify(Object object);

    <T> T parse(String json, Class<T> clazz);

}

File: src/main/java/de/byjoker/jfql/connection/Connection.kt
package de.byjoker.jfql.connection

import de.byjoker.jfql.statement.Statement
import de.byjoker.jfql.util.Response

interface Connection {
    fun connect()
    fun disconnect()
    fun closeSession(ignoreStaticSessions: Boolean = true)
    fun query(query: String, exception: Boolean = true): Response
    fun query(statement: Statement, exception: Boolean = true): Response
    fun isConnected(): Boolean
}
